test/task_test.rb

```diff
@@ -0,0 +1,98 @@
+using_task_library "iodrivers_base"
+require "iodrivers_base/orogen_test_helpers"
+
+require_relative "modbus_helpers.rb"
+
+using_task_library "water_probe_acquanativa_ap3"
+import_types_from "water_probe_acquanativa_ap3"
+
+
+describe OroGen.water_probe_acquanativa_ap3.Task do
+    include IODriversBase::OroGenTestHelpers
+
+    run_live
+
+    attr_reader :task
+    attr_reader :reader
+    attr_reader :writer
+
+    include ModbusHelpers
+
+    def mock_all_sensor_registers
+        registers = {
+            "dissolved_oxygen" => 0,
+            "dissolved_oxygen_sat" => 1,
+            "temperature" => 2,
+            "ph" => 3,
+            "conductivity" => 4,
+            "salinity" => 5,
+            "dissolved_solids" => 6,
+            "specific_gravity" => 7,
+            "orp" => 8,
+            "turbity" => 9,
+            "height" => 10,
+            "latitude" => 11,
+            "longitude" => 12
+        }
+
+        max_value = 2**16
+        modbus_set(registers["dissolved_oxygen"], rand(max_value))
+        modbus_set(registers["dissolved_oxygen_sat"], rand(max_value))
+        modbus_set(registers["temperature"], rand(max_value))
+        modbus_set(registers["ph"], rand(max_value))
+        modbus_set(registers["conductivity"], rand(max_value))
+        modbus_set(registers["salinity"], rand(max_value))
+        modbus_set(registers["dissolved_solids"], rand(max_value))
+        modbus_set(registers["specific_gravity"], rand(max_value))
+        modbus_set(registers["orp"], rand(max_value))
+        modbus_set(registers["turbity"], rand(max_value))
+        modbus_set(registers["height"], rand(max_value))
+        modbus_set(registers["latitude"], rand(max_value))
+        modbus_set(registers["longitude"], rand(max_value))
```

Comment:
You should put chosen values and check for them in the "successfully read all measurements" test

Reply:
but the library checks this values in https://github.com/tidewise/drivers-water_probe_acquanativa_ap3/ ... we thoght that the component does not write the values, it just requires data from the output. Should we change it?

Reply:
update: Talked with @doudou, and he thinks that the component should check the values also. Maybe not as strictly as the lib, but at least some.